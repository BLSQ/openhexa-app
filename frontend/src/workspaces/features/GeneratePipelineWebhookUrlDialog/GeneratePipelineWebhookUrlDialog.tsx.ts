import { graphql } from "graphql/gql";

export const GeneratePipelineWebhookUrlDialogPipelineDoc = graphql(`
fragment GeneratePipelineWebhookUrlDialog_pipeline on Pipeline {
  id
  code
}
`);
