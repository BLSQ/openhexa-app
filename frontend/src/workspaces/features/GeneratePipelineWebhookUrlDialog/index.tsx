import GeneratePipelineWebhookUrlDialog from "./GeneratePipelineWebhookUrlDialog";

export default GeneratePipelineWebhookUrlDialog;
