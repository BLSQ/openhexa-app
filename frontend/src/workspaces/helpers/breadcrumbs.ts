/**
 * Utility functions for handling breadcrumb navigation in bucket file systems
 */

export interface BreadcrumbItem {
  label: string;
  value: string;
}

/**
 * Generates breadcrumb items from a prefix path
 * @param prefix - The current prefix path (e.g., "folder1/folder2/")
 * @returns Array of breadcrumb items with label and value for navigation
 */
export function generateBreadcrumbs(prefix: string | null): BreadcrumbItem[] {
  if (!prefix) return [];

  const breadcrumbs: BreadcrumbItem[] = [];
  let accumulated = "";

  prefix
    .split("/")
    .filter(Boolean)
    .forEach((part) => {
      accumulated = accumulated ? `${accumulated}/${part}` : part;
      breadcrumbs.push({
        label: part,
        value: `${accumulated}/`,
      });
    });

  return breadcrumbs;
}
