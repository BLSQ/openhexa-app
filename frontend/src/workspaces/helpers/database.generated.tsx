import * as Types from '../../graphql/types';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type DeleteWorkspaceDatabaseTableMutationVariables = Types.Exact<{
  input: Types.DeleteWorkspaceDatabaseTableInput;
}>;


export type DeleteWorkspaceDatabaseTableMutation = { __typename?: 'Mutation', deleteWorkspaceDatabaseTable?: { __typename?: 'DeleteWorkspaceDatabaseTableResult', success: boolean, errors: Array<Types.DeleteWorkspaceDatabaseTableError> } | null };


export const DeleteWorkspaceDatabaseTableDocument = gql`
    mutation deleteWorkspaceDatabaseTable($input: DeleteWorkspaceDatabaseTableInput!) {
  deleteWorkspaceDatabaseTable(input: $input) {
    success
    errors
  }
}
    `;
export type DeleteWorkspaceDatabaseTableMutationFn = Apollo.MutationFunction<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>;

/**
 * __useDeleteWorkspaceDatabaseTableMutation__
 *
 * To run a mutation, you first call `useDeleteWorkspaceDatabaseTableMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useDeleteWorkspaceDatabaseTableMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [deleteWorkspaceDatabaseTableMutation, { data, loading, error }] = useDeleteWorkspaceDatabaseTableMutation({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useDeleteWorkspaceDatabaseTableMutation(baseOptions?: Apollo.MutationHookOptions<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>(DeleteWorkspaceDatabaseTableDocument, options);
      }
export type DeleteWorkspaceDatabaseTableMutationHookResult = ReturnType<typeof useDeleteWorkspaceDatabaseTableMutation>;
export type DeleteWorkspaceDatabaseTableMutationResult = Apollo.MutationResult<DeleteWorkspaceDatabaseTableMutation>;
export type DeleteWorkspaceDatabaseTableMutationOptions = Apollo.BaseMutationOptions<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>;