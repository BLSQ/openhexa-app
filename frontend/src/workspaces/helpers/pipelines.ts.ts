import { graphql } from "graphql/gql";

export const NewRunDoc = graphql(`
fragment NewRun on PipelineRun {
  id
}
`);
