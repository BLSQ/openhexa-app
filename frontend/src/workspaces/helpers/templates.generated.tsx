import * as Types from '../../graphql/types';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type UpdateWorkspaceTemplateMutationVariables = Types.Exact<{
  input: Types.UpdateTemplateInput;
}>;


export type UpdateWorkspaceTemplateMutation = { __typename?: 'Mutation', updatePipelineTemplate: { __typename?: 'UpdateTemplateResult', success: boolean, errors: Array<Types.UpdateTemplateError>, template?: { __typename?: 'PipelineTemplate', id: string, name: string, description?: string | null, config?: string | null } | null } };

export type DeleteTemplateVersionMutationVariables = Types.Exact<{
  input: Types.DeleteTemplateVersionInput;
}>;


export type DeleteTemplateVersionMutation = { __typename?: 'Mutation', deleteTemplateVersion: { __typename?: 'DeleteTemplateVersionResult', success: boolean, errors: Array<Types.DeleteTemplateVersionError> } };


export const UpdateWorkspaceTemplateDocument = gql`
    mutation UpdateWorkspaceTemplate($input: UpdateTemplateInput!) {
  updatePipelineTemplate(input: $input) {
    success
    errors
    template {
      id
      name
      description
      config
    }
  }
}
    `;
export type UpdateWorkspaceTemplateMutationFn = Apollo.MutationFunction<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>;

/**
 * __useUpdateWorkspaceTemplateMutation__
 *
 * To run a mutation, you first call `useUpdateWorkspaceTemplateMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useUpdateWorkspaceTemplateMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [updateWorkspaceTemplateMutation, { data, loading, error }] = useUpdateWorkspaceTemplateMutation({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useUpdateWorkspaceTemplateMutation(baseOptions?: Apollo.MutationHookOptions<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>(UpdateWorkspaceTemplateDocument, options);
      }
export type UpdateWorkspaceTemplateMutationHookResult = ReturnType<typeof useUpdateWorkspaceTemplateMutation>;
export type UpdateWorkspaceTemplateMutationResult = Apollo.MutationResult<UpdateWorkspaceTemplateMutation>;
export type UpdateWorkspaceTemplateMutationOptions = Apollo.BaseMutationOptions<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>;
export const DeleteTemplateVersionDocument = gql`
    mutation DeleteTemplateVersion($input: DeleteTemplateVersionInput!) {
  deleteTemplateVersion(input: $input) {
    success
    errors
  }
}
    `;
export type DeleteTemplateVersionMutationFn = Apollo.MutationFunction<DeleteTemplateVersionMutation, DeleteTemplateVersionMutationVariables>;

/**
 * __useDeleteTemplateVersionMutation__
 *
 * To run a mutation, you first call `useDeleteTemplateVersionMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useDeleteTemplateVersionMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [deleteTemplateVersionMutation, { data, loading, error }] = useDeleteTemplateVersionMutation({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useDeleteTemplateVersionMutation(baseOptions?: Apollo.MutationHookOptions<DeleteTemplateVersionMutation, DeleteTemplateVersionMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<DeleteTemplateVersionMutation, DeleteTemplateVersionMutationVariables>(DeleteTemplateVersionDocument, options);
      }
export type DeleteTemplateVersionMutationHookResult = ReturnType<typeof useDeleteTemplateVersionMutation>;
export type DeleteTemplateVersionMutationResult = Apollo.MutationResult<DeleteTemplateVersionMutation>;
export type DeleteTemplateVersionMutationOptions = Apollo.BaseMutationOptions<DeleteTemplateVersionMutation, DeleteTemplateVersionMutationVariables>;