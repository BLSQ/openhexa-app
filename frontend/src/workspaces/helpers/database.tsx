import { gql } from "@apollo/client";
import { getApolloClient } from "core/helpers/apollo";
import {
  DeleteWorkspaceDatabaseTableMutation,
  DeleteWorkspaceDatabaseTableMutationVariables,
} from "./database.generated";

type lang = "R" | "PYTHON";

export const getReadTableSnippet = (tableName: string) => {
  return `import os
import pandas as pd
from sqlalchemy import create_engine

engine = create_engine(os.environ["WORKSPACE_DATABASE_URL"])

pd.read_sql("SELECT * FROM ${tableName}", con=engine)`;
};

export const getUsageSnippet = (tableName: string, lang: lang) => {
  let text = "";
  switch (lang) {
    case "PYTHON":
      text = `import os
import pandas as pd
from sqlalchemy import create_engine

engine = create_engine(os.environ["WORKSPACE_DATABASE_URL"])

# Create sample dataframe
df = pd.DataFrame({"name": ["Jane", "John", "Tyler"], "age": [19, 17, 22]})

# Write data
df.to_sql("${tableName}", con=engine, if_exists="replace")`;
      break;
    case "R":
      text = `library(DBI)
  
con <- dbConnect(
    RPostgres::Postgres(),
    dbname = Sys.getenv("WORKSPACE_DATABASE_DB_NAME"),
    host = Sys.getenv("WORKSPACE_DATABASE_HOST"),
    port = Sys.getenv("WORKSPACE_DATABASE_PORT"),
    user = Sys.getenv("WORKSPACE_DATABASE_USERNAME"),
    password = Sys.getenv("WORKSPACE_DATABASE_PASSWORD")
)

dbWriteTable(con, "${tableName}", Data_fin, overwrite=TRUE)`;
      break;
  }
  return text;
};

export async function deleteTable(workspace: string, tableName: string) {
  const client = getApolloClient();
  const { data } = await client.mutate<
    DeleteWorkspaceDatabaseTableMutation,
    DeleteWorkspaceDatabaseTableMutationVariables
  >({
    mutation: gql`
      mutation deleteWorkspaceDatabaseTable(
        $input: DeleteWorkspaceDatabaseTableInput!
      ) {
        deleteWorkspaceDatabaseTable(input: $input) {
          success
          errors
        }
      }
    `,
    variables: { input: { workspaceSlug: workspace, table: tableName } },
  });

  if (!data?.deleteWorkspaceDatabaseTable?.success) {
    throw new Error("Impossible to delete table");
  }
  return true;
}
