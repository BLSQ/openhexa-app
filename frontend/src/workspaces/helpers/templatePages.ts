import { GetServerSidePropsContext, GetServerSidePropsResult } from "next";
import { CustomApolloClient } from "core/helpers/apollo";
import {
  WorkspaceTemplatePageDocument,
  WorkspaceTemplatePageQuery,
  WorkspaceTemplatePageQueryVariables,
} from "workspaces/graphql/queries.generated";
import TemplateLayout from "workspaces/layouts/TemplateLayout";

type TemplatePageProps = {
  workspaceSlug: string;
  templateCode: string;
};

export const createTemplatePageServerSideProps = () => ({
  requireAuth: true,
  async getServerSideProps(
    ctx: GetServerSidePropsContext,
    client: CustomApolloClient,
  ): Promise<GetServerSidePropsResult<TemplatePageProps>> {
    await TemplateLayout.prefetch(ctx, client);
    const { data } = await client.query<
      WorkspaceTemplatePageQuery,
      WorkspaceTemplatePageQueryVariables
    >({
      query: WorkspaceTemplatePageDocument,
      variables: {
        workspaceSlug: ctx.params!.workspaceSlug as string,
        templateCode: ctx.params!.templateCode as string,
      },
    });

    if (!data.workspace || !data.template) {
      return { notFound: true };
    }
    return {
      props: {
        workspaceSlug: ctx.params!.workspaceSlug as string,
        templateCode: ctx.params!.templateCode as string,
      },
    };
  },
});
