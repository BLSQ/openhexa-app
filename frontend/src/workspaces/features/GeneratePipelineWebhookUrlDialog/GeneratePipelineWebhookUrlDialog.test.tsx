import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TestApp } from "core/helpers/testutils";
import GeneratePipelineWebhookUrlDialog from ".";
import { faker } from "@faker-js/faker";

const mutationMock = jest.fn();

jest.mock("@apollo/client", () => ({
  ...jest.requireActual("@apollo/client"),
  useMutation: jest.fn().mockImplementation(() => [mutationMock]),
}));

const PIPELINE = {
  id: faker.string.uuid(),
  code: faker.string.alphanumeric(),
};

describe("GeneratePipelineWebhookUrlDialog", () => {
  const onClose = jest.fn();
  beforeEach(() => {
    mutationMock.mockClear();
  });

  it("is displayed when open is true", async () => {
    const { container } = render(
      <TestApp mocks={[]}>
        <GeneratePipelineWebhookUrlDialog
          pipeline={PIPELINE}
          open={true}
          onClose={() => {}}
        />
      </TestApp>,
    );

    const dialog = await screen.queryByRole("dialog");
    expect(dialog).toBeInTheDocument();
    expect(onClose).not.toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  it("Generates new webhook url", async () => {
    mutationMock.mockReturnValue({
      data: { generatePipelineWebhookUrl: { errors: [], success: true } },
    });

    const user = userEvent.setup();

    const { container } = render(
      <GeneratePipelineWebhookUrlDialog
        pipeline={PIPELINE}
        open={true}
        onClose={() => {}}
      />,
    );

    const saveButton = screen.getByRole("button", { name: "Generate" });
    await user.click(saveButton);

    expect(mutationMock).toHaveBeenCalled();
  });
});
