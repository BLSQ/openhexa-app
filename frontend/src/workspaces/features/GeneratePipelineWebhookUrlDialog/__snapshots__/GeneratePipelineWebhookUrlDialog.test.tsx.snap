// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneratePipelineWebhookUrlDialog is displayed when open is true 1`] = `
<div
  aria-hidden="true"
/>
`;
